code/app/src/main/java/com/example/sigma_blue/Tag.java

```diff
@@ -0,0 +1,55 @@
+package com.example.sigma_blue;
+
+import android.graphics.Color;
+
+/**
+ * Stores information about a single tag.
+ */
+public class Tag implements Comparable<Tag>{
+    private String tagText;
+    private Color colour;
+
+    public Tag(String tagText, int colour) {
+        this.tagText = tagText;
+        //this.colour = Color.valueOf(colour); // requires API 26
+    }
+
+    public Tag(String tagText, Color colour) {
+        this.tagText = tagText;
+        this.colour = colour;
+    }
+
+    public String getTagText() {
+        return tagText;
+    }
+
+    public void setTagText(String tagText) {
+        this.tagText = tagText;
+    }
+
+    public Color getColour() {
+        return colour;
+    }
+
+    @Override
+    public int compareTo(Tag o) {
+        return this.tagText.compareTo(o.getTagText());
+    }
+
+    @Override
+    public int hashCode() {
+        return this.tagText.hashCode();
+    }
+
+    @Override
```

Comment:
Should we do a color comparison as well? Should that be left for future implementation?

Reply:
I think having two tags with the same name but different colours is a bad use case.